app/controllers/subjects_controller.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+# Subjects contoller for list, update, destroy etc
+class SubjectsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_subject_for_destroy, only: %i[destroy]
+  before_action :find_subject, only: %i[show update]
+
+  def index
+    @subjects = Subject.all
+    render json: {
+      status: { code: 200, message: 'Success.' },
+      data: SubjectSerializer.new(@subjects).serializable_hash[:data]
+    }
+  end
+
+  def show
+    render_success_response('success.', true)
+  end
+
+  def create
+    @subject = Subject.new(subject_params)
+    return unless authorize @subject
+
+    if @subject.save
+      render_success_response('success.', true)
+    else
+      render_failed_response('Failed to create subject.')
+    end
+  end
+
+  def update
+    return unless authorize @subject
+
+    begin
+      @subject.update(subject_params)
+      render_success_response('Successfully updated the subject.', true)
+    rescue Exception => e
+      render_failed_response(e.message)
+    end
+  end
+
+  def destroy
```

Comment:
We are calling private method "find_subject_for_destroy" before destroy method. So it will try to find the subject with that ID.
If it is find or not will always return true and success message.

Reply:
okay. Looks good. Thanks